addons/loyalty/models/loyalty_reward.py

```diff
@@ -241,7 +241,10 @@ def write(self, vals):
 
     def unlink(self):
         programs = self.program_id
-        res = super().unlink()
+        if self.env['sale.order.line'].search_count([('reward_id.id', '=', self.id)]):
+            res = self.action_archive()
+        else:
+            res = super().unlink()
```

Comment:
`loyalty` doesn't depend on `sale`, you cannot search on `sale.order.line` here.

Also, could you add a test to cover the flow you're trying to fix ? Thanks